File: trunk/HeartRateMonitor.Server/HeartRateMonitor.Math/Evaluation/Geometry/EvaluateBasicHistogram.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace HeartRateMonitor.Math.Evaluation.Geometry
{
    public class EvaluateBasicHistogram : IEvaluation<Histogram>
    {
        public Histogram Evaluate(SessionData session)
        {
            var h = new Histogram().Init();
            foreach (int interval in session.Intervals) {
                h.AddRRInterval(interval);
            }
            return h;
        }
    }
}

File: trunk/PolarMath/src/org/macondo/polar/evaluation/geometry/EvaluateBasicHistogram.java
package org.macondo.polar.evaluation.geometry;

import org.macondo.polar.evaluation.Evaluation;
import org.macondo.polar.data.Training;

import java.util.List;

/**
 * <p></p>
 *
 * Date: 21.05.2008
 *
 * @author <a href="mailto:ktsibriy@gmail.com">Kirill Y. Tsibriy</a>
 */
public class EvaluateBasicHistogram implements Evaluation<Histogram> {
    public Histogram evaluate(Training training) {
        Histogram h = new Histogram().init();
        List<Integer> intervals = training.getIntervals();
        for (Integer interval : intervals) {
            h.addRRInterval(interval);
        }
        return h;
    }
}
